packages/@react-aria/focus/src/useFocusRing.ts

```diff
@@ -37,7 +37,14 @@ export function useFocusRing(props: FocusRingProps = {}): FocusRingAria {
   let {within} = props;
   let [isFocused, setFocused] = useState(false);
   let [isFocusWithin, setFocusWithin] = useState(false);
-  let {isFocusVisible} = useFocusVisible(props);
+  let [isFocusVisible, setFocusVisible] = useState(false);
+
+  // trigger on isFocusVisible state change when component is focused and modality changes
+  // or when the component loses focus
+  useFocusEmitter(isFocusVisible => {
+    setFocusVisible((within ? isFocusWithin : isFocused) && isFocusVisible);
+  }, [within, isFocusWithin, isFocused]);
```

Comment:
I think this needs a bit more logic inside it to ensure parity with previous behavior, specifically the part regarding text inputs that `useFocusVisible` handled:
```
    if (isTextInput && modality === 'keyboard' && e instanceof KeyboardEvent && !FOCUS_VISIBLE_INPUT_KEYS[e.key]) {
      return;
    }
```
This snippet of code made it so that the focus ring wouldn't show up if a user clicks into a textfield and began typing/using arrow keys to move the cursor. Right now doing the same flow causes the focus ring to appear. Currently trying some stuff to get this to work.

Repo steps just in case:
1. Go to textfield story
2. Click into the textfield. Note the absence of a focus ring (aka bold blue ring)
3. Type in field. Note the focus ring appears.
4. Try this in https://reactspectrum.blob.core.windows.net/reactspectrum/2f9bc85a761db55a56eb43bc06dbbefd5f019589/storybook/index.html?path=/story/textfield--default (older storybook). Note that the focus ring won't appear in step 3

Reply:
Looks like a quick fix would be to copy paste that logic and put it before the `setFocusVisible` call, will need to either export or make a copy of FOCUS_VISIBLE_INPUT_KEYS.